src/detect-http-ua.c

```diff
@@ -178,6 +181,29 @@ static InspectionBuffer *GetData(DetectEngineThreadCtx *det_ctx,
     return buffer;
 }
 
+static InspectionBuffer *GetData2(DetectEngineThreadCtx *det_ctx,
```

Comment:
Do we want to have a generic function for this ?
Having a callback as an argument to get the buffer and its length